audience_strategy.go

```diff
@@ -46,8 +46,46 @@ func DefaultAudienceMatchingStrategy(haystack []string, needle []string) error {
 	return nil
 }
 
+func ExactAudienceMatchingStrategy(haystack []string, needle []string) error {
```

Comment:
Maybe add a small codedoc to explain why we need this and when it will be used? I think you could use the text from the PR description and maybe add one example when this would be neccessary.

Reply:
Added.